cli/init.js

```diff
@@ -18,11 +20,18 @@ module.exports = function init(args) {
   const relativePublicDir = path.relative(CWD, absolutePublicDir)
   const dirExists = fs.existsSync(absolutePublicDir)
 
-  invariant(
-    dirExists,
-    'Failed to create a Service Worker at "%s": directory does not exist.\nMake sure to include a relative path to the root directory of your server.',
-    absolutePublicDir,
-  )
+  if (!dirExists) {
+    // Try to create the directory if it doesn't exist
+    const [createDirectoryError] = await until(() =>
+      fs.promises.mkdir(absolutePublicDir, { recursive: true }),
+    )
+    invariant(
+      createDirectoryError == null,
+      `Failed to create a Service Worker at "%s": directory does not exist and could not be created.${EOL}Make sure to include a relative path to the root directory of your server.${EOL}${EOL}See the original error below:${EOL}%s`,
```

Comment:
Why prefer `EOL` over `\n`?

Reply:
Because it might differ on windows vs linux/mac
![Screenshot 2021-04-27 at 14 29 15](https://user-images.githubusercontent.com/7705041/116241260-ff7ecb80-a764-11eb-8b9a-70f681a134e7.png)

Reply:
I think it's a good change, but it doesn't belong to this pull request's scope. 

What do you think if we use `\n` here, as we do across the codebase, and fix them _all_ in another pull request? That way changes are scoped, clear, and revertable.

Reply:
Sure, I'll put it back